packages/devtools_app/lib/src/service/vm_service_wrapper.dart

```diff
@@ -285,9 +290,57 @@ class VmServiceWrapper implements VmService {
         },
       });
     }
+
+    await _addProcessedUrlsToTraceObject(
+        isolateId, resolvedUrlMap, traceObject);
+
     return CpuProfileData.parse(traceObject);
   }
 
+  // Helper function for fetching simplified package urls for each of
+  // the [resolvedUrls] then mapping them to the [traceObject].
+  //
+  // [isolateId] The id which is passed to the getIsolate RPC to load this isolate.
+  // [resolvedUrlMap] A map of where each key is a resolved url that has been extracted from the [traceObject]'s
+  // stack frames. The values of the map will be overriden in this helper.
+  // This is passed as a parameter since [resolvedUrls] can be extracted
+  // while [traceObject] is being created.
+  // [traceObject] A map where the CpuProfileData for each frame is stored.
+  Future<void> _addProcessedUrlsToTraceObject(
+      String isolateId,
+      Map<String, String?> resolvedUrlMap,
+      Map<String, dynamic> traceObject) async {
+    final processedUrlMapping = <String, String>{};
+    final resolvedUrlList = resolvedUrlMap.keys.toList();
+    final uris =
+        (await _vmService.lookupPackageUris(isolateId, resolvedUrlList)).uris;
```

Comment:
piggy backing off of my last comment, we should only request the package uri for resolved urls we do not already have a mapped package uri for. We'll probably want to check for this where `resolvedUrlMap` is built

Reply:
ah right saving this in the wrapper service context allows us to persist those over a much longer time so we can squeek some efficiency out!